src/source-af-packet.c

```diff
@@ -2284,12 +2291,12 @@ TmEcode AFPSetBPFFilter(AFPThreadVars *ptv)
  *
  * \param mapfd file descriptor of the protocol bypass table
  * \param key data to use as key in the table
- * \param inittime time of creation of the entry (in monotonic clock)
  * \return 0 in case of error, 1 if success
  */
-static int AFPInsertHalfFlow(int mapd, void *key, uint64_t inittime)
+static int AFPInsertHalfFlow(int mapd, void *key, uint32_t hash,
```

Comment:
Not related to this patch set, but still things I want to mention:

1. use  ```} else if (PKT_IS_IPV6(p) &&```
2. this is wrong ```((IPV6_GET_NH(p) == IPPROTO_TCP) || (IPV6_GET_NH(p) == IPPROTO_UDP))) {``` I think. At the func start you already checked if the packet is TCP or UDP. Using IPV6_GET_NH is almost always wrong.
3. you can avoid all the htonl calls by directly accessing the packet ipv4 and ipv6 headers
4. I'm curious about the per call checks for ```p->afp_v.v4_map_fd```. Can we just check those at start up? Perhaps create different wrappers for the Callback in case v4 is available and v6 isn't?